carbonserver/render.go

```diff
@@ -744,33 +737,56 @@ func (listener *CarbonserverListener) Render(req *protov2.MultiFetchRequest, str
 		key, size := listener.getRenderCacheKeyAndSize(targets, format.String()+"grpc")
 		var res interface{}
 		cacheT0 := time.Now()
-		res, fromCache, err = getWithCache(logger, listener.queryCache, key, size, 60,
-			func() (interface{}, error) {
-				return fetchAndStreamMetricsFunc(true)
-			})
-		tle.CacheDuration = float64(time.Since(cacheT0)) / float64(time.Second)
-		if err == nil {
-			listener.prometheus.cacheRequest("query", fromCache)
-			if fromCache {
-				atomic.AddUint64(&listener.metrics.QueryCacheHit, 1)
-				cacheResponse := res.([]response)
+
+		item := listener.queryCache.getQueryItem(key, size, 60)
+		res, cacheOk := item.FetchOrLock()
```

Comment:
It wasn't clear for me what `cacheOk` means here, after reading FetchOrLock() code I think it means something like "gotValue". If so, maybe worth renaming the variable to make it more intuitive.

Reply:
It's about whether the results were found in the cache or not.
So changed to `found`.